drivers/serial/uart_nrfx_uart.c

```diff
@@ -64,6 +64,7 @@ static inline struct uart_nrfx_data *get_dev_data(struct device *dev)
 
 #ifdef CONFIG_UART_0_ASYNC
 static struct {
+	struct device *dev;
```

Comment:
since this is single instance driver you could add `DEVICE_DECLARE(uart_nrfx_uart0)` at the top and use `DEVICE_GET()` instead of creating `struct device *dev` field.

Reply:
sure, done